control-plane/csi-driver/src/bin/node/registration.rs

```diff
@@ -0,0 +1,37 @@
+use crate::client::AppNodesClientWrapper;
+use snafu::Snafu;
+use std::{collections::HashMap, time::Duration};
+use tokio::task::JoinError;
+use tracing::error;
+
+#[derive(Debug, Snafu)]
+pub(crate) enum RegistrationError {
+    TokioTaskWait { source: JoinError },
+}
+
+/// Default registration interval.
+const REGISTRATION_INTERVAL_ON_SUCCESS: Duration = Duration::from_secs(60 * 5);
+/// Default registration interval on error.
+const REGISTRATION_INTERVAL_ON_ERROR: Duration = Duration::from_secs(30);
+
+pub(crate) async fn run_registration_loop(
+    id: String,
+    endpoint: String,
+    labels: Option<HashMap<String, String>>,
+    client: &AppNodesClientWrapper,
+) {
+    let mut logged_error = false;
+    loop {
+        let interval_duration = match client.register_app_node(&id, &endpoint, &labels).await {
+            Ok(_) => REGISTRATION_INTERVAL_ON_SUCCESS,
```

Comment:
There's a bug here, logged_error needs to be reset on successful registration.
```suggestion
            Ok(_) => {
                if logged_error {
                    tracing::info!("Successfully re-registered the app_node");
                    logged_error = false;
                }
                REGISTRATION_INTERVAL_ON_SUCCESS
            }
```

Reply:
Oops, right.